### PR TITLE
Adds license and security vulnerabilities checks for Hadoop3 build

### PR DESCRIPTION
**Pull request by tejaswini-imply**
Hadoop 2 often causes red security scans on druid builds because of the dependencies it brings. We want to move away from Hadoop 2 and have to provide Hadoop 3 distribution available.

New stages are added in the pipeline to make the license and security-vulnerabilities checks that run with Hadoop2 also run with the Hadoop3 build.

P.S. Not updating Hadoop3 IT tests pipelines since Hadoop tests are not running on Travis currently.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -142,27 +151,10 @@ jobs:
...
-        If there are false positive dependency analysis warnings, they can be suppressed:
-        https://maven.apache.org/plugins/maven-dependency-plugin/analyze-mojo.html#usedDependencies
-        https://maven.apache.org/plugins/maven-dependency-plugin/examples/exclude-dependencies-from-dependency-analysis.html
-
-        For more information, refer to:
-        https://maven.apache.org/plugins/maven-dependency-plugin/analyze-mojo.html
-
-        " && false; }
+    - <<: *analyze_dependencies
+      name: "analyze hadoop 3 dependencies"
```
Comment (cryptoe) [inline .travis.yml:155]: Can we introduce a dummy code change and role it back to see if `analyze hadoop 3 dependencies` works. 
We can roll it back prior to merge.

**Review by cryptoe — COMMENTED** [review]
Looks like `https://app.travis-ci.com/github/apache/druid/jobs/587017381` has not run since there is not code change. Can we make it run by introducing a code change, add a empty line and then roll the change back prior to merging.

File: .travis.yml (hunk 1 above)
Comment (abhishekagarwal87) [reply .travis.yml:155]: https://app.travis-ci.com/github/apache/druid/jobs/587017381 - does this work?

Comment (tejaswini-imply) [reply .travis.yml:155]: I can also add analyze dependencies, analyze hadoop 3 dependencies to check_test_suite.py and then revert it back.

**Comment by tejaswini-imply** [comment]
@cryptoe Could you please review this now, analyze dependencies and analyze hadoop3 dependencies seem to be working as expected. Previously a new line crept into the mvn command.

**Review by abhishekagarwal87 — APPROVED** [review]

**Pull request merged**